### PR TITLE
Add missing steps to characteristic's WriteValue function

### PR DESCRIPTION
**Pull request by zakorgy**
Add two missing steps to characteristic's WriteValue() function. https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-writevalue (Step 4 and 5)

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because, there are no webbluetooth test api implementation yet.

### CODE REVIEW

File: components/script/dom/bluetoothremotegattcharacteristic.rs [hunk 1]
```diff
@@ -180,6 +183,12 @@ impl BluetoothRemoteGATTCharacteristicMethods for BluetoothRemoteGATTCharacteris
         if uuid_is_blacklisted(self.uuid.as_ref(), Blacklist::Writes) {
             return Err(Security)
         }
+        if value.len() > MAXIMUM_ATTRIBUTE_LENGTH {
+            return Err(NoModificationAllowed)
+        }
+        if !self.Service().Device().Gatt().Connected() {
```
Comment (mmatyas) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:190]: I suppose `Service()`, `Device()` and `Gatt()` always return a valid value, right?

Comment (zakorgy) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:190]: yes

**Comment by mmatyas** [comment]
ok, r+

File: components/script/dom/bluetoothremotegattcharacteristic.rs [hunk 2]
```diff
@@ -23,6 +23,9 @@ use dom::bluetoothuuid::{BluetoothDescriptorUUID, BluetoothUUID};
 use ipc_channel::ipc::{self, IpcSender};
 use net_traits::bluetooth_thread::BluetoothMethodMsg;
 
+// Maximum length of an attribute value.
+const MAXIMUM_ATTRIBUTE_LENGTH: usize = 512;
```
Comment (mmatyas) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:28]: Is this documented somewhere?

Comment (zakorgy) [inline components/script/dom/bluetoothremotegattcharacteristic.rs:28]: https://www.bluetooth.org/DocMan/handlers/DownloadDoc.ashx?doc_id=286439 (Vol. 3, page 2169) and you can find it in the specification steps too: https://webbluetoothcg.github.io/web-bluetooth/#dom-bluetoothremotegattcharacteristic-writevalue (Step 4).

**Pull request closed without merging**